packages/rax/src/markupChecksum.js

```diff
@@ -0,0 +1,23 @@
+const CHECKSUM_ATTR_NAME = 'data-rax-checksum';
+const TAG_END = /\/?>/;
+const CHECKSUM = 'rax-checksum';
+
+/**
+  * add checksum to markup
+*/
+export function addChecksumToMarkup(markup) {
+  markup = markup || '';
+
+  // Add checksum (handle both parent tags, comments and self-closing tags)
+  return `<div ${CHECKSUM_ATTR_NAME}="${CHECKSUM}">${markup}</div>`;
```

Comment:
rax can return a node list, instead of  adding checksum to each node. use a div wrap them and add checksum to the div. in browser, check the div instead of checking children in the div.